apps/full-site-editing/full-site-editing-plugin/starter-page-templates/class-starter-page-templates.php

```diff
@@ -118,19 +118,33 @@ public function enqueue_assets() {
 			return;
 		}
 
-		// Load Tracks data if available.
-		$tracks_identity    = null;
-		$is_wpcom           = ( defined( 'IS_WPCOM' ) && IS_WPCOM );
-		$has_active_jetpack = ( class_exists( 'Jetpack' ) && Jetpack::is_active() );
-		if ( $has_active_jetpack && ! $is_wpcom && class_exists( 'Jetpack_Tracks_Client' ) ) {
-			$tracks_identity = Jetpack_Tracks_Client::get_connected_user_tracks_identity();
-			wp_enqueue_script(
-				'jp-tracks',
-				'//stats.wp.com/w.js',
-				array(),
-				gmdate( 'YW' ),
-				true
-			);
+		// Enable tracking only on specific platforms.
+		$tracks_identity = null;
+		$is_atomic       = defined( 'IS_ATOMIC' ) && IS_ATOMIC;
+		$is_wpcom        = defined( 'IS_WPCOM' ) && IS_WPCOM;
+		if ( $is_atomic || $is_wpcom ) {
+			// Load identity.
+			$has_active_jetpack = ( class_exists( 'Jetpack' ) && Jetpack::is_active() );
+			if ( $has_active_jetpack && class_exists( 'Jetpack_Tracks_Client' ) ) {
+				$tracks_identity = Jetpack_Tracks_Client::get_connected_user_tracks_identity();
+			} elseif ( $is_wpcom ) {
+				$user_id         = get_current_user_id();
```

Comment:
This can return `0`. Unlikely or impossible on WPCom but if it _did_ happen then `get_user_by( 'id', $user_id )->user_login,` would error.

Reply:
what do you recommend? maybe this?

```php
'username' => $user_id ? get_user_by( 'id', $user_id )->user_login : null,
```

Reply:
> This can return 0.

Is there a way to reproduce that?

Reply:
Not really. Just aware it _could_ happen in theory.